test/resources/ns_with_lots_of_meta.clj

```diff
@@ -1,12 +1,18 @@
-(ns ^:md1 ^:md2 ^{:longhand "as well" :really? "yes" :ok true} ns-with-gen-class-methods-meta
+(ns ^:md1 ^:md2 ^{:longhand "as well" :really? "yes" :ok true}
+    ns-with-gen-class-methods-meta
   (:gen-class
    :methods [^:static [foo [String] String]
              ^{:test true} [bar [String] String]
              ^{:other "text"} [baz [String] String]]
    :name Name)
-  (:require [clojure.string :as s]
-            [clojure.pprint :refer [fresh-line]]))
+  (:require
+   [clojure.string :as s]
+   [clojure.pprint :refer [fresh-line]]))
 
 (defn useit
   []
   (fresh-line))
+
+(defn issue-239-metadata-based-protocol-extension []
+  ;; https://github.com/clojure-emacs/refactor-nrepl/issues/239
+  ^{`s/whatever-goes true} {})
```

Comment:
Could be nice to also exercise a case with metametadata:

```clj
(defn foo ^{:foo ^:bar []} [])
```

I think this PR already handles that as I see some recursion, so maybe no impl changes will be even needed

Reply:
Yeah, it will handle that, but the stack will blow if you go all Inception on your source code with meta levels 🙄